TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java

```diff
@@ -0,0 +1,530 @@
+
+public class InteroperabilityPerformanceTest {
+
+    private static ProcessBuilder pb;
+
+    private final String json = "application/json";
+
+    private final int[] targetSize = {1,10,50,100,200,500,1000};
+
+    private final OperatingSystemMXBean osBean = (OperatingSystemMXBean) ManagementFactory.getOperatingSystemMXBean();;
+
+    @BeforeAll
+    static void setupProcess() {
+        String os = System.getProperty("os.name");
+        if (os.toLowerCase().contains("windows")) {
+            pb = new ProcessBuilder(
+                    "cmd.exe", "/c", "java -jar .\\src\\test\\resources\\runTodoManagerRestAPI-1.5.5.jar");
+        }
+        else {
+            pb = new ProcessBuilder(
+                    "sh", "-c", "java -jar ./src/test/resources/runTodoManagerRestAPI-1.5.5.jar");
+        }
+    }
+
+    @BeforeEach
+    void startServer() throws InterruptedException {
+        try {
+            pb.start();
+            Thread.sleep(1000);
+        } catch (IOException e) {
+            System.out.println("No server");
+        }
+    }
+
+    @AfterEach
+    void shutServer() {
+        try {
+            RestAssured.get("http://localhost:4567/shutdown");
+        }
+        catch (Exception ignored) {
+        }
+    }
+
+    @Test
+    public void AddCategoriesToTodo() {
+        int max = targetSize[targetSize.length-1];
+        String todoId;
+        String[] categoryId = new String[max];
+        double[] time = new double[targetSize.length];
+        double[] cpuUsage = new double[targetSize.length];
+        long[] freeMemory = new long[targetSize.length];
+
+        // Get a new todoitem
+        Response response = RestAssured.given().body("{\"title\":\"todo\"}").post("http://localhost:4567/todos");
+        todoId = response.jsonPath().get("id");
+
+        // Generate categories, store the id
+        for(int i = 0; i < max; i++) {
+            String body = "{\"title\":\"Test\",\"description\":\"\"}";
+            response = RestAssured.given()
+                    .header("Accept", json)
+                    .body(body)
+                    .post("http://localhost:4567/categories");
+            categoryId[i] = response.jsonPath().get("id");
+        }
+
+        // Add categories to todoitem
+        String url = "http://localhost:4567/todos/" + todoId + "/categories";
+        int targetIndex = 0;
+        for (int j = 1; j <= max; j++) {
+            long startTime = System.nanoTime();
+            String body = "{\"id\":\"" + categoryId[j-1] + "\"}";
+            RestAssured.given()
+                    .header("Accept", json)
+                    .body(body)
+                    .post(url);
+            long endTime = System.nanoTime();
+            if (j == targetSize[targetIndex]) {
+                double cpuLoad = osBean.getCpuLoad() * 100;
+                long memory = osBean.getFreeMemorySize() / (1024^2);
+                time[targetIndex] = (double) (double) (endTime-startTime) / (10^9);
```

Comment:
issue with time, not correct for exponential operator, you have to something like Math.pow(1024, 2) or just 1024 * 1024

Reply:
done